logging/logrus/logger.go

```diff
@@ -280,9 +286,9 @@ func (logger *Logger) GetLevel() logging.LogLevel {
 
 // AddHook adds a hook to the standard logger hooks.
 func (logger *Logger) AddHook(hook lg.Hook) {
-	logger.access.Lock()
-	defer logger.access.Unlock()
+	std := logger.stdStore.Load().(*lg.Logger)
```

Comment:
It is not probable to add too many hooks therefore you can apply same solution as for Formatter - think about making the Hooks effectively immutable (that it will be same situation as for the Formatter).

https://stackoverflow.com/questions/16678416/different-between-immutable-and-effectively-immutable-objects